File: _Src/Freem/Freem.Web.Api.Public.Contracts/DTO/ErrorResponse.cs
namespace Freem.Web.Api.Public.Contracts.DTO;

public sealed class ErrorResponse
{
    public string Code { get; }
    
    public string Message { get; }
    public IReadOnlyDictionary<string, string>? Properties { get; }
    
    public ErrorResponse(string code, string message, IReadOnlyDictionary<string, string>? properties = null)
    {
        Code = code;
        Message = message;
        Properties = properties;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/ErrorResponse.ts
export class ErrorResponse {
  public readonly type: string;
  public readonly message?: string;
  public readonly properties?: { [key: string]: string };

  constructor(type: string, message?: string, properties?: { [key: string]: string }) {
    this.type = type;
    this.message = message;
    this.properties = properties;
  }
}